File: C++/Sorting/452. Minimum Number of Arrows to Burst Balloons.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
    static bool compare(vector<int> &a, vector<int> &b) {
        return a[1] < b[1];
    }

public:
    int findMinArrowShots(vector<vector<int>> &points) {
        sort(points.begin(), points.end(), compare);
        int result = 1, prev = points[0][1];
        for (int i = 1; i < points.size(); ++i) {
            if (points[i][0] <= prev) continue;
            result++, prev = points[i][1];
        }
        return result;
    }
};

File: Python/sorting/452. Minimum Number of Arrows to Burst Balloons.py
from typing import List


class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        res, end = 0, float('-inf')
        for start, cur_end in points:
            if start > end:
                res += 1
                end = cur_end
        return res